MinuteRaid/GlobalLighting.gd

```diff
@@ -0,0 +1,54 @@
+extends CanvasModulate
+
+const DAY_COLOR = Color(0.97, 0.93, 0.68, 1)
+const NIGHT_COLOR = Color(0.16, 0.16, 0.12, 1)
+
+var _elapsed = 0
+
+func _ready():
+	set_color(hsv_lerp(DAY_COLOR, NIGHT_COLOR, 0.5))
+
+func _process(delta):
+	# Want to transition from day to night every minute
+	_elapsed += delta
+	_elapsed = fmod(_elapsed,120.0)
+	_shift_global_lighting(_elapsed)
+		
+func _shift_global_lighting(elapsed):
+	var _interpolation = 0.0
+	if elapsed >= 30.0 && elapsed < 90.0:
+		_interpolation = (elapsed - 30.0)/60.0
+		set_color(hsv_lerp(DAY_COLOR, NIGHT_COLOR, _interpolation))
+	elif elapsed >= 90.0:
+		_interpolation = (elapsed - 90.0)/60.0
+		set_color(hsv_lerp(NIGHT_COLOR, DAY_COLOR, _interpolation))
+	else:
+		_interpolation = (elapsed + 30.0)/60.0
+		set_color(hsv_lerp(NIGHT_COLOR, DAY_COLOR, _interpolation))
+		
+func hsv_lerp(cola, colb, t):
+	#This part will flip the direction of the lerp if the two colors are above
+	 #180 degrees apart, this way the lerp always takes the shortest path.
+	var h
+	var ha = cola.h
```

Comment:
These variable names scream at me 😆

Reply:
yeah, I can comment/clean this up a bit more